src/classes/WavetableCache.tsx

```diff
@@ -0,0 +1,40 @@
+import Wavetable from "./Wavetable";
+import wavetables from "../data/wavetables";
+
+export default class WavetableCache {
+    static singleton: WavetableCache;
+    cache: Map<string, Wavetable>;
+
+    constructor() {
+        this.cache = new Map<string, Wavetable>();
+    }
+
+    static getSingleton() {
+        if (WavetableCache.singleton == null) {
```

Comment:
stylistic nit - the more JS way of doing this would be `if(!WavetableCache.singleton)`